### PR TITLE
Document inline arrays.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/types.rst [hunk 1]
```diff
@@ -356,6 +356,44 @@ the ``.length`` member.
...
+
+    contract C {
+        function f() {
+            // The next line creates a type error because uint[3] memory
+            // cannot be converted to uint[] memory.
+            uint[] x = [uint(1), 3, 4];
+    }
+
+It is planned to remove this restriction in the future but currently creates
+some complications because of how arryes are passed in the ABI.
```
Comment (Denton-L) [inline docs/types.rst]: Typo, "arryes"

Comment (chriseth) [inline docs/types.rst]: Thanks, fixed.

**Comment by dmdque** [comment]
Re: how it isn't possible to assign an inline array to a dynamically-sized memory array

> It is planned to remove this restriction in the future

Is this being tracked anywhere? It would be a great improvement to Solidity!

**Comment by chriseth** [comment]
It turned out to be more complicated, unfortunately, but we should still try.

**Pull request merged**